src/AaveV2_Eth_ServiceProviders_20231907/COWSwapper20230726.sol

```diff
@@ -17,70 +17,87 @@ contract COWSwapper {
       CHAINLINK_PRICE_CHECKER,
       _getEncodedData(AaveV2EthereumAssets.USDT_ORACLE, AaveV2EthereumAssets.USDC_ORACLE)
     );
 
     IMilkman(MILKMAN).requestSwapExactTokensForTokens(
-      adaiBalance,
-      IERC20(AaveV2EthereumAssets.DAI_A_TOKEN),
-      IERC20(AaveV2EthereumAssets.USDC_A_TOKEN),
+      daiBalance,
+      IERC20(AaveV2EthereumAssets.DAI_UNDERLYING),
+      IERC20(AaveV2EthereumAssets.USDC_UNDERLYING),
       address(AaveV2Ethereum.COLLECTOR),
       CHAINLINK_PRICE_CHECKER,
       _getEncodedData(AaveV2EthereumAssets.DAI_ORACLE, AaveV2EthereumAssets.USDC_ORACLE)
     );
   }
 
-  function cancelSwap(address ausdtMilkman, address adaiMilkman) external {
+  function cancelUsdt(address milkman) external {
     if (msg.sender != ALLOWED_CALLER && msg.sender != AaveGovernanceV2.SHORT_EXECUTOR) {
       revert InvalidCaller();
     }
 
-    IMilkman(ausdtMilkman).cancelSwap(
-      ausdtBalance,
-      IERC20(AaveV2EthereumAssets.USDT_A_TOKEN),
-      IERC20(AaveV2EthereumAssets.USDC_A_TOKEN),
+    IMilkman(milkman).cancelSwap(
+      usdtBalance,
+      IERC20(AaveV2EthereumAssets.USDT_UNDERLYING),
+      IERC20(AaveV2EthereumAssets.USDC_UNDERLYING),
       address(AaveV2Ethereum.COLLECTOR),
       CHAINLINK_PRICE_CHECKER,
       _getEncodedData(AaveV2EthereumAssets.USDT_ORACLE, AaveV2EthereumAssets.USDC_ORACLE)
     );
 
-    IMilkman(adaiMilkman).cancelSwap(
-      adaiBalance,
-      IERC20(AaveV2EthereumAssets.DAI_A_TOKEN),
-      IERC20(AaveV2EthereumAssets.USDC_A_TOKEN),
+    IERC20(AaveV2EthereumAssets.USDT_UNDERLYING).safeTransfer(
       address(AaveV2Ethereum.COLLECTOR),
-      CHAINLINK_PRICE_CHECKER,
-      _getEncodedData(AaveV2EthereumAssets.DAI_ORACLE, AaveV2EthereumAssets.USDC_ORACLE)
+      IERC20(AaveV2EthereumAssets.USDT_UNDERLYING).balanceOf(address(this))
     );
+  }
+
+  function cancelDai(address milkman) external {
+    if (msg.sender != ALLOWED_CALLER && msg.sender != AaveGovernanceV2.SHORT_EXECUTOR) {
+      revert InvalidCaller();
+    }
 
-    IERC20(AaveV2EthereumAssets.WETH_UNDERLYING).safeTransfer(
+    IMilkman(milkman).cancelSwap(
+      daiBalance,
+      IERC20(AaveV2EthereumAssets.DAI_UNDERLYING),
+      IERC20(AaveV2EthereumAssets.USDC_UNDERLYING),
       address(AaveV2Ethereum.COLLECTOR),
-      IERC20(AaveV2EthereumAssets.WETH_UNDERLYING).balanceOf(address(this))
+      CHAINLINK_PRICE_CHECKER,
+      _getEncodedData(AaveV2EthereumAssets.DAI_ORACLE, AaveV2EthereumAssets.USDC_ORACLE)
     );
-    IERC20(AaveV2EthereumAssets.BAL_UNDERLYING).safeTransfer(
+
+    IERC20(AaveV2EthereumAssets.DAI_UNDERLYING).safeTransfer(
       address(AaveV2Ethereum.COLLECTOR),
-      IERC20(AaveV2EthereumAssets.BAL_UNDERLYING).balanceOf(address(this))
+      IERC20(AaveV2EthereumAssets.DAI_UNDERLYING).balanceOf(address(this))
     );
   }
 
+  function depositIntoAaveV2(address token) external {
+    if (msg.sender != ALLOWED_CALLER && msg.sender != AaveGovernanceV2.SHORT_EXECUTOR) {
```

Comment:
i think this could be made permissionless

Reply:
@rustboyar that's a good point, i've updated, thanks.